### PR TITLE
Intel OpenVINO backend (inference only)

### LINKED ISSUE #13987: Intel OpenVINO inference backend
# 🚀 Feature request

Hi all! We (Intel team) would like to contribute in Hugging Face by integration of optimized deep learning inference framework OpenVINO. According to the contribution guidelines, we should discuss it with community and developers to match the expectations. Let me share what we have as this moment:

* New format for the [models hub](https://huggingface.co/) - a pair of `ov_model.xml` and `ov_model.bin` which is OpenVINO IR
* New class `OVAutoModel` which downloads `ov_model.xml/.bin` by default or framework specific checkpoint (PyTorch, TensorFlow) with on-the-fly conversion. Example:
  ```python
  from transformers import OVAutoModel
  model = OVAutoModel.from_pretrained("dkurt/test_openvino")
  ```
  (see https://huggingface.co/dkurt/test_openvino)
  or
  ```python
  # Downloads PyTorch weights and converts to OpenVINO in runtime
  model = OVAutoModel.from_pretrained("albert-base-v2", from_pt=True)
  ```
  or
  ```python
  # Downloads TensorFlow weights and converts to OpenVINO in runtime
  model = OVAutoModel.from_pretrained("distilbert-base-uncased", from_tf=True)
  ```

## Motivation

Expected efficiency improvement as well as utilization of Intel hardware (CPU/iGPU/VPU and others supported)

## Your contribution

Will do our best to complete this feature request. Looking forward for feedback from developers and experienced users.

### PR DESCRIPTION
**Pull request by dkurt**
# What does this PR do?

* This PR adds an optional [OpenVINO](https://github.com/openvinotoolkit/openvino) backend which allows to run deep learning inference (not training) on Intel Hardware (CPUs, GPUs, VPUs and other).
* OpenVINO can load models in ONNX or OpenVINO IR formats. Using one of the following classes, conversion is done in runtime (with `from_pt` or `from_tf` flags, see the tests for details).
```
OVAutoModel
OVAutoModelForMaskedLM
OVAutoModelForQuestionAnswering
OVAutoModelWithLMHead
OVAutoModelForSequenceClassification
```
* Users might upload IR format directly to the hub. In example, https://huggingface.co/dkurt/bert-large-uncased-whole-word-masking-squad-int8-0001

forum: https://discuss.huggingface.co/t/intel-openvino-backend/11178  
resolves https://github.com/huggingface/transformers/issues/13987

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by LysandreJik** [comment]
Hello @dkurt, as you have seen with @mfuntowicz and @echarlaix, and as you have opened a PR in optimum already, I'll go ahead and close this. It will still be visible and the code will still be accessible, in case reopening it in the future makes sense.

Thank you for your contribution!
